### PR TITLE
Support System tables at PostgreSQLClient

### PR DESCRIPTION
**Pull request by miniway**

### CODE REVIEW

File: presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java [hunk 1]
```diff
@@ -112,6 +113,13 @@ public void testForeignTable()
         execute("DROP SERVER devnull");
     }
 
+    @Test
+    public void testSystemTable()
+            throws Exception
+    {
+        assertGreaterThan(computeActual("SHOW TABLES FROM pg_catalog").getRowCount(), 0);
```
Comment (findepi) [inline presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java]: - verify that `SHOW TABLES FROM pg_catalog` returned some well-known names (just 2-3 examples, not all)
- add a test for reading from such a table `SELECT * FROM pg_catalog.xxxx`

File: presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java [hunk 2]
```diff
@@ -200,7 +200,7 @@ protected ResultSet getTables(Connection connection, Optional<String> schemaName
                 connection.getCatalog(),
                 escapeNamePattern(schemaName, escape).orElse(null),
                 escapeNamePattern(tableName, escape).orElse(null),
-                new String[] {"TABLE", "VIEW", "MATERIALIZED VIEW", "FOREIGN TABLE"});
+                new String[] {"TABLE", "VIEW", "MATERIALIZED VIEW", "FOREIGN TABLE", "SYSTEM TABLE", "SYSTEM VIEW"});
```
Comment (findepi) [inline presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java]: pg_catalog.xxx are SYSTEM TABLEs, VIEWs or mix?

**Comment by findepi** [comment]
If someone was having an access control for postgresql catalog, then exposing `pg_catalog` may be a security issue.

Maybe we do this with a flag?

@electrum @kokosing

**Comment by electrum** [comment]
Having a flag for this sounds like a good idea. We should probably default to false by default, as it could be a potential security issue.

File: presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java (hunk 2 above)
Comment (miniway) [reply presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java]: pg_catalog.xxx are mix of them and it's not easy to tell simply by their names also. ex, `pg_foreign_table` is SYSTEM TABLE but `pg_tables` is SYSTEM VIEW

**Comment by miniway** [comment]
Thanks for the review, I've modified the test case.
Also I'm going to add a flag at `PostgreSqlConfig` to include SYSTEM TABLE and SYSTEM VIEW

**Comment by miniway** [comment]
@findepi I’ve added a flag for system tables

File: presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java [hunk 3]
```diff
@@ -200,7 +210,7 @@ protected ResultSet getTables(Connection connection, Optional<String> schemaName
                 connection.getCatalog(),
                 escapeNamePattern(schemaName, escape).orElse(null),
                 escapeNamePattern(tableName, escape).orElse(null),
-                new String[] {"TABLE", "VIEW", "MATERIALIZED VIEW", "FOREIGN TABLE"});
+                tableTypes);
```
Comment (findepi) [inline presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java]: `String[]` is not immutable:
```suggestion
                tableTypes.clone());
```

File: presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java [hunk 4]
```diff
@@ -135,6 +138,13 @@ public PostgreSqlClient(
         this.jsonType = typeManager.getType(new TypeSignature(StandardTypes.JSON));
         this.uuidType = typeManager.getType(new TypeSignature(StandardTypes.UUID));
         this.varcharMapType = (MapType) typeManager.getType(TypeSignature.parseTypeSignature("map(varchar, varchar)"));
+        String[] tableTypes = new String[] {"TABLE", "VIEW", "MATERIALIZED VIEW", "FOREIGN TABLE"};
+        if (postgreSqlConfig.isIncludeSystemTables()) {
+            tableTypes = Stream.concat(
+                    Arrays.stream(tableTypes),
```
Comment (findepi) [inline presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java]: The init code is a bit complex. What about this:
```
List<String> tableTypes = new ArrayList<>();
addAll(tableTypes, "TABLE", "VIEW", "MATERIALIZED VIEW", "FOREIGN TABLE"); // Collections.addAll
if (postgreSqlConfig.isIncludeSystemTables()) {
  addAll(tableTypes, "SYSTEM TABLE", "SYSTEM VIEW");
}
this.tableTypes = tableTypes.toArray(new String[0]);
```

**Comment by miniway** [comment]
@findepi Thanks for the comments. Rebased and squashed

**Comment by findepi** [comment]
Merged as 87ce835b700a1444cf841b5ccaa73c52f1446258, thanks!

**Comment by findepi** [comment]
@miniway as a follow-up, would you like to mention this in https://prestosql.io/docs/current/connector/postgresql.html ?

**Pull request closed without merging**